### PR TITLE
Added Qstash api reference for different languages

### PR DESCRIPTION
**Pull request by BrightonMboya**
In this PR, I have added the API reference for different languages (js, python, and Go) for a better dx when referencing the api. The proposal was accepted on Discord.

### CODE REVIEW

File: qstash/api/events/list.mdx [hunk 1]
```diff
@@ -65,11 +65,47 @@ The name of the endpoint if this message was sent through a topic
...
   -H "Authorization: Bearer <token>"
 ```
 
+```javascript Node
+fetch('https://qstash.upstash.io/v2/events', {
+  headers: {
+    'Authorization': 'Bearer <token>'
+  }
+});
+```
```
Comment (mdumandag) [inline qstash/api/events/list.mdx:79]: To make the examples consistent with each other, I think we can do this. Same applies to other node.js snippets
```suggestion
```javascript Node
const response = await fetch('https://qstash.upstash.io/v2/events', {
  headers: {
    'Authorization': 'Bearer <token>'
  }
});
```
```

File: qstash/api/events/list.mdx [hunk 2]
```diff
@@ -65,11 +65,47 @@ The name of the endpoint if this message was sent through a topic
...
+if err != nil {
+  log.Fatal(err)
+}
+defer resp.Body.Close()
+bodyText, err := io.ReadAll(resp.Body)
+if err != nil {
+  log.Fatal(err)
+}
+fmt.Printf("%s\n", bodyText) 
+```
```
Comment (mdumandag) [inline qstash/api/events/list.mdx:107]: To make the examples consistent with each other, I think we can do this. Same applies to other go snippets
```suggestion
```go Go
req, err := http.NewRequest("GET", "https://qstash.upstash.io/v2/events", nil)
if err != nil {
  log.Fatal(err)
}
req.Header.Set("Authorization", "Bearer <token>")
resp, err := http.DefaultClient.Do(req)
if err != nil {
  log.Fatal(err)
}
defer resp.Body.Close()
```
```

File: qstash/api/messages/create.mdx [hunk 3]
```diff
@@ -108,20 +108,83 @@ curl -X POST "https://qstash.upstash.io/v2/publish/https://www.example.com" \
...
+    'Upstash-Delay': '10s',
+    'Upstash-Retries': '3',
+    'Upstash-Forward-Custom-Header': 'custom-value',
+}
+
+json_data = {
+    'message': 'Hello, World!',
+}
+
+response = requests.post('https://qstash.upstash.io/v2/publish/https://www.example.com', headers=headers, json=json_data)
```
Comment (mdumandag) [inline qstash/api/messages/create.mdx:169]: Minor: To make the lines horizontally shorter, you can do something like this
```suggestion
response = requests.post(
    'https://qstash.upstash.io/v2/publish/https://www.example.com', 
    headers=headers, 
    json=json_data,
)
```

**Comment by BrightonMboya** [comment]
@mdumandag have corrected the go and Node snippets as requested

File: qstash/api/messages/create.mdx [hunk 4]
```diff
@@ -108,20 +108,84 @@ curl -X POST "https://qstash.upstash.io/v2/publish/https://www.example.com" \
...
+      "Upstash-Forward-Custom-Header": "custom-value",
+    },
+    body: JSON.stringify({
+      message: "Hello, World!",
+    }),
+  }
+);
+```
+
+```go Go
```
Comment (mdumandag) [inline qstash/api/messages/create.mdx:131]: Can you make the order as curl, Node, Python, Go here to be consistent with the rest?

File: qstash/api/messages/get.mdx [hunk 5]
```diff
@@ -17,11 +17,45 @@ authMethod: "bearer"
 
 <RequestExample>
 
-```sh
+```sh curl
 curl https://qstash.upstash.io/v2/messages/msg_123 \
   -H "Authorization: Bearer <token>"
 ```
 
+```js node
```
Comment (mdumandag) [inline qstash/api/messages/get.mdx:25]:
```suggestion
```js Node
```

File: qstash/api/messages/get.mdx [hunk 6]
```diff
@@ -17,11 +17,45 @@ authMethod: "bearer"
...
+const response = await fetch("https://qstash.upstash.io/v2/messages/msg_123", {
+  headers: {
+    Authorization: "Bearer <token>",
+  },
+});
+```
+
+```go Go
```
Comment (mdumandag) [inline qstash/api/messages/get.mdx:33]: Can you make the order as curl, Node, Python, Go here to be consistent with the rest?

File: qstash/api/topics/add-endpoint.mdx [hunk 7]
```diff
@@ -51,4 +51,82 @@ curl -XPOST https://qstash.upstash.io/v2/topics/:topicName/endpoints \
...
+        'name': 'endpoint2',
+        'url': 'https://somewhere-else.com'
+      }
+    ]
+  })
+});
+```
+
+
+```python python
```
Comment (mdumandag) [inline qstash/api/topics/add-endpoint.mdx:77]:
```suggestion
```python Python
```

**Review by mdumandag — COMMENTED** [review]
Looks good to me with minor comments

**Comment by BrightonMboya** [comment]
hopefully it will be good to go

**Review by mdumandag — APPROVED** [review]
Thanks a lot @BrightonMboya, it all looks good to me now

**Pull request merged**